operators/bosh-restore-operator/DefaultBoshRestoreOperator.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const logger = require('../../common/logger');
+const eventmesh = require('../../data-access-layer/eventmesh');
+const catalog = require('../../common/models/catalog');
+const CONST = require('../../common/constants');
+const BaseOperator = require('../BaseOperator');
+const BoshRestoreService = require('./');
+
+class DefaultBoshRestoreOperator extends BaseOperator {
+
+  init() {
+    const RESTORE_STATES = [
+      `${CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS}_BOSH_STOP`,
+      `${CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS}_CREATE_DISK`,
+      `${CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS}_ATTACH_DISK`,
+      `${CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS}_PUT_FILE`,
+      `${CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS}_RUN_ERRANDS`,
+      `${CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS}_BOSH_START`,
+      `${CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS}_POST_BOSH_START`
+    ];
+    const defaultValidStatelist = [
+      CONST.APISERVER.RESOURCE_STATE.IN_QUEUE
+    ];
+    const validStateList = defaultValidStatelist.concat(RESTORE_STATES);
+    return this.registerCrds(CONST.APISERVER.RESOURCE_GROUPS.RESTORE, CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BOSH_RESTORE)
+      .then(() => this.registerWatcher(CONST.APISERVER.RESOURCE_GROUPS.RESTORE, CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BOSH_RESTORE, validStateList));
+  }
+
+  async processRequest(requestObjectBody) { 
+    try {
```

Comment:
I would have a timeout time for restore operation. For whatsoever reason the restore operation is not finishing within the restore SLO, then the operation must be marked as failed and should not be processed. Ex., post 36 hrs or 48 hrs after restore start if restore is yet to be processed (could be stuck in any phase), such operations must be timed out.

Reply:
@jagadish-kb : We will add this after the services have come up with a baseline for their timings. Hope that's fine